runtime/onert/core/src/ir/pass/PermutationInsertionPass.cc

```diff
@@ -92,25 +92,27 @@ void PermutationInsertionPass::callback(const OperandIndex &index, Operand &obje
       const backend::Backend *backend = op_seq_li->backend();
       assert(backend);
       auto use_node_inputs = operation.getInputs();
-      // Controlflow operation can have the same operand as 2 or more inputs
-      if (backend->config()->id() != backend::controlflow::Config::ID)
-      {
-        assert(use_node_inputs.contains(index));
-      }
+      assert(use_node_inputs.contains(index));
 
       auto new_index = factor_to_index.at({backend, op_seq_layout});
       if (index != new_index)
       {
         // Update from op_seq
-        _lowered_graph.op_seqs().at(op_seq_index).replaceInput(index, new_index);
+        // Replace the same inputs of an OpSequence at once for the following reasons:
+        // 1. An OpSequence's inputs are the same inputs of first operation
+        // 2. An OpSequence may have inputs as the same operand (2 or more).
+        // 3. The same inputs of OpSequence have the same PermuteFactor.
```

Comment:
@ragmani It seems the assumption (`assumed invariant`). Then, it may be better written with `assert` than `comment`. i) to find out when it is invalid, ii) c++ statement  is usually more clear than natural language.

Reply:
I don't know how to add assertion these reasons Could you recommend which assertion you thought?